src/crunker.ts

```diff
@@ -0,0 +1,359 @@
+export interface CrunkerConstructorOptions {
+  /**
+   * Sample rate for Crunker's internal audio context.
+   *
+   * @default 44100
+   */
+  sampleRate: number;
+}
+
+export type CrunkerInputTypes = string | File;
+
+/**
+ * An exported Crunker audio object.
+ */
+export interface ExportedCrunkerAudio {
+  blob: Blob;
+  url: string;
+  element: HTMLAudioElement;
+}
+
+/**
+ * Crunker is the simple way to merge, concatenate, play, export and download audio files using the Web Audio API.
+ */
+export default class Crunker {
+  private readonly _sampleRate: number;
+  private readonly _context: AudioContext;
+
+  /**
+   * Creates a new instance of Crunker with the provided options.
+   */
+  constructor({ sampleRate = 44100 }: Partial<CrunkerConstructorOptions> = {}) {
+    this._sampleRate = sampleRate;
+    this._context = this._createContext();
+  }
+
+  /**
+   * Creates Crunker's internal AudioContext.
+   *
+   * @internal
+   */
+  private _createContext(): AudioContext {
+    window.AudioContext = window.AudioContext || (window as any).webkitAudioContext || (window as any).mozAudioContext;
+    return new AudioContext();
+  }
+
+  /**
+   * Asynchronously fetches multiple audio files and returns an array of AudioBuffers.
+   */
+  async fetchAudio(...filepaths: CrunkerInputTypes[]): Promise<AudioBuffer[]> {
+    const files = filepaths.map(async (filepath) => {
+      let buffer: ArrayBuffer;
+
+      if (filepath instanceof File) {
+        buffer = await filepath.arrayBuffer();
+      } else {
+        buffer = await fetch(filepath).then((response) => response.arrayBuffer());
+      }
+
+      return await this._context.decodeAudioData(buffer);
+    });
+
+    return await Promise.all(files);
+  }
+
+  /**
+   * Merges (layers) multiple AudioBuffers into a single AudioBuffer.
+   *
+   * **Visual representation:**
+   *
+   * ![](https://user-images.githubusercontent.com/12958674/88806278-968f0680-d186-11ea-9cb5-8ef2606ffcc7.png)
+   */
+  mergeAudio(buffers: AudioBuffer[]): AudioBuffer {
+    const output = this._context.createBuffer(
+      this._maxNumberOfChannels(buffers),
+      this._sampleRate * this._maxDuration(buffers),
+      this._sampleRate
+    );
+
+    buffers.forEach((buffer) => {
+      for (let channelNumber = 0; channelNumber < buffer.numberOfChannels; channelNumber += 1) {
```

Comment:
I'm wondering if we should change the `let` declarations in all the for-loops to `const` since we don't need to redeclare and const is more "strict" for these things.

Reply:
I don't think that would work though, as the final statement in the for loop increments the variable.

Reply:
All good, my bad, only gleamed over the code yesterday. Doing some testing locally now. Thanks for the additional minor code quality commits.